src/pdf_parser/pdf_utils/postprocess_layout.py

```diff
@@ -0,0 +1,305 @@
+import itertools
```

Comment:
Most of this file is trivially refactored from the old classes. But the "infer_missing_blocks_from_gaps" function has been refactored to fill gaps at the start/end of pages.